### PR TITLE
[CI][E2E] Optimize CI E2E test performance

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

The E2E tests of the current CI are executed in parallel, and Maven is compiled repeatedly during each test. Under the limited CI server resources, the server load will be very high, which may lead to the instability of E2E test.

This PR mainly changes multiple Maven compilations to one. So that E2E tests executed in parallel only need to pull the docker image to complete the test. This will greatly improve the stability of E2E test.

### CODE REVIEW

**Comment by SbloodyS** [comment]
@caishunfeng @zhongjiajie @kezhenxu94 Please review. Thanks.

**Comment by caishunfeng** [comment]
well done, it's a good idea.

**Comment by kezhenxu94** [comment]
> ## Purpose of the pull request
> The E2E tests of the current CI are executed in parallel, and Maven is compiled repeatedly during each test. Under the limited CI server resources, the server load will be very high, which may lead to the instability of E2E test.

To correct a nit. The parallel tests are running in different CI runners (VMs)

But still this is a good improvement. Thanks!

**Review by kezhenxu94 — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -27,9 +27,44 @@ concurrency:
...
+          docker save apache/dolphinscheduler-standalone-server:ci -o /tmp/standalone-image.tar \
+          && du -sh /tmp/standalone-image.tar
+      - uses: actions/upload-artifact@v2
+        name: Upload Docker Images
+        with:
+          name: standalone-image
+          path: /tmp/standalone-image.tar
+          retention-days: 1
   e2e:
     name: ${{ matrix.case.name }}
```
Comment (zhongjiajie) [inline .github/workflows/e2e.yml:66]: It's odd here, I think the case of the matrix is running in a different virtual machine, Am I wrong? If they running in the different machines, then each of them should run and build a standalone docker image whether we add additional github action step or not. Did I miss something here? @kezhenxu94

Comment (zhongjiajie) [reply .github/workflows/e2e.yml:66]: I think this patch only makes total times of the E2E test shorter, but it still would lack of resource in E2E as before

Comment (SbloodyS) [reply .github/workflows/e2e.yml:66]: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
```
You can specify the runner type for each job in a workflow. Each job in a workflow executes in a fresh instance of the virtual machine. All steps in the job execute in the same instance of the virtual machine, allowing the actions in that job to share information using the filesystem.
```

This document only explains the difference between steps and jobs, but does not explain the difference between matrix. I am also very confused about that.

Comment (kezhenxu94) [reply .github/workflows/e2e.yml:66]: Jobs are always running in different machines.

**Review by zhongjiajie — APPROVED** [review]
+1, Thank for clarification

**Pull request merged**